src/main/java/domain/AttendanceSheets.java

```diff
@@ -0,0 +1,142 @@
+            throw new IllegalArgumentException("[ERROR] 등록되지 않은 닉네임입니다.");
+        }
+    }
+
+    public void add(AttendanceSheet attendanceSheet) {
+        validateIsAlreadyAttendance(attendanceSheet);
+        attendanceSheets.add(attendanceSheet);
+    }
+
+    private void validateIsAlreadyAttendance(AttendanceSheet attendanceSheet) {
+        if (attendanceSheets.stream()
+                .anyMatch(sheet -> sheet.equals(attendanceSheet))) {
+            throw new IllegalArgumentException("[ERROR] 이미 출석했습니다. 수정 기능을 이용하세요.");
+        }
+    }
+
+    public AttendanceSheet findAttendanceSheetByNicknameAndDay(String nickname, int day) {
+        List<AttendanceSheet> attendanceByNickname = findAttendanceByNickname(nickname);
+
+        return attendanceByNickname.stream().filter(sheet -> sheet.isSameDay(day)).findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("[ERROR] 출석 기록이 없는 날짜입니다."));
+    }
+
+    public List<AttendanceSheet> findAttendanceByNickname(String nickname) {
+        validateNullOrEmptyNickname(nickname);
+
+        List<AttendanceSheet> foundAttendance = attendanceSheets.stream()
+                .filter(attendanceSheet -> attendanceSheet.hasNickname(nickname))
+                .toList();
+
+        validateExistAttendanceSheetsByNickname(foundAttendance);
+
+        return foundAttendance;
+    }
+
+    public List<String> findAllNames() {
+        return attendanceSheets.stream().map(AttendanceSheet::getNickname).distinct().toList();
+    }
+
+    public AttendanceStatics calculateRiskOfExpulsionBy(String nickname, LocalDate today) {
+        int lateCount = calculateLateCountBy(nickname);
+        int absentCount = calculateAbsentCount(nickname, today);
+
+        return new AttendanceStatics(0, lateCount, absentCount);
+    }
+
+    public AttendanceStatics calculateAttendanceStaticsBy(String nickname, LocalDate today) {
+        int attendCount = calculateAttendCountBy(nickname);
+        int lateCount = calculateLateCountBy(nickname);
+        int absentCount = calculateAbsentCount(nickname, today);
+
+        return new AttendanceStatics(attendCount, lateCount, absentCount);
+    }
+
+    public int calculateAttendCountBy(String nickname) {
+        return calculateCountByNicknameAndState(nickname, AttendanceState.ATTEND);
+    }
+
+    public int calculateLateCountBy(String nickname) {
+        return calculateCountByNicknameAndState(nickname, AttendanceState.LATE);
+    }
+
+    private int calculateCountByNicknameAndState(String nickname, AttendanceState state) {
+        List<AttendanceSheet> attendanceByNickname = findAttendanceByNickname(nickname);
+        int count = 0;
+
+        for (AttendanceSheet attendanceSheet : attendanceByNickname) {
+            AttendanceDateTime attendanceDateTime = attendanceSheet.getAttendanceDateTime();
+
+            count += countStateByAttendanceDateTime(attendanceDateTime, state);
+        }
+
+        return count;
```

Comment:
Stream으로 변환 가능할 것 같습니다!

```suggestion
        return (int) attendanceByNickname.stream()
                .map(AttendanceSheet::getAttendanceDateTime)
                .filter(attendanceDateTime -> attendanceDateTime.check() == state)
                .count();
```
```